python/tests/test_equality.py

```diff
@@ -211,7 +211,7 @@ def test_generic_equality():
 
     assert a == b
 
-    a = tm.MyTuple(v1=42.0, v2="hello, world")
+    a = tm.MyTuple[float, str](v1=42.0, v2="hello, world")
```

Comment:
Interesting that we lose type inference here

Reply:
I agree. I'm not sure yet if we can make the type inference work for aliased generic Python types. Let me look into right now.

FYI it's not a regression, just something that wasn't previously tested. On  the current main branch (31a6e29), we get the same Pylance/Pyright error if I add the following line to the pytests (since AliasedTuple = MyTuple:
 
 `a = tm.AliasedTuple(v1=42.0, v2="hello, world")`

`Argument of type "float" cannot be assigned to parameter "v1" of type "T1@AliasedTuple" in function "__init__"
  Type "float" cannot be assigned to type "T1@AliasedTuple"Pylance[reportGeneralTypeIssues]`

Reply:
It looks like we can drop the type parameters when declaring an alias to a type in the same module, but not across modules.

Reply:
I don't think the location (i.e. module) of the declaration matters. I just pushed another commit that fixes this.

If the alias declaration (NamedType) is generic and resolves to a Record, then we can drop the type parameters.

Aliasing the name instead of creating a python `TypeAlias` makes the type checker inference happy. I understand why aliasing the name works, but I don't understand why the type checkers can't infer type arguments when you create a proper TypeAlias!

```python
class Tuple(Generic[T1, T2]):
    v1: T1
    v2: T2
    ...

# Creating a TypeAlias is technically correct as far as I understand the Python docs
AliasedTuple: TypeAlias = Tuple[T1, T2]

# But type checkers can't infer type arguments and complains here. Weak
t1 = AliasedTuple(v1="hello", v2=42)

# Creating a new alias symbol works because they are the *same* type
RenamedTuple: type = Tuple

# No problem for type checker type inference
t2 = RenamedTuple(v1="hello", v2=42)
```

So I suppose we'll go with the way that works.